packages/react-import-remote/README.md

```diff
@@ -0,0 +1,135 @@
+
+Provides a utility function and React component to asynchronously load scripts and extract the newly-added global
+
+## Installation
+
+```bash
+$ yarn add @shopify/react-import-remote
+```
+
+## Usage
+
+The provided utilities are intended only for external scripts that load globals. Other JavaScript should use the native `import()` operator for asynchronously loading code. These utilities cache results by source, so only a single `script` tag is ever added for a particular source.
+
+### `<ImportRemote />`
+
+A React component that loads the script specified in the `source` prop.
+
+```ts
+import * as React from 'react';
+import ImportRemote from '@shopify/react-import-remote';
+
+interface RemoteGlobal {}
+interface WindowWithGlobal extends Window {
+  remoteGlobal: RemoteGlobal;
+}
+
+class MyComponent extends React.Component {
+  ...
+
+  render() {
+    ...
+
+    return (
+      <ImportRemote
+        preload
+        source="https://some-external-service.com/global.js"
+        getImport={(window: WindowWithGlobal) => window.remoteGlobal}
+        onError={(error: Error) => this.setState({error})}
+        onImported={(remoteGlobal: RemoteGlobal) => doSomethingWithGlobal(remoteGlobal)}
+      />
+    );
+  }
+}
+```
+
+### `load()`
+
+A more primitive utility function for asynchronously loading remote scripts. Simply provide the source and a function that takes the newly-augmented window and returns the global from that script.
+
+```ts
+import {load} from '@shopify/react-import-remote';
+
+interface RemoteGlobal {}
+interface WindowWithGlobal extends Window {
+  remoteGlobal: RemoteGlobal;
+}
+
+const source = 'https://some-external-service.com/global.js';
+
+function getImport(window: WindowWithGlobal) {
+  return window.remoteGlobal;
+}
+
+// Without using async await
+load(source, getImport)
+  .then(imported => {
+    doSomething(imported);
+  })
+  .catch(error => {
+    doSomethingWithError(error);
+  });
+
+// using ES8 async await
+try {
+  const imported = await load(source, getImport);
+  doSomething(imported);
+} catch (error) {
+  doSomethingWithError(error);
+}
+```
+
+## Interfaces
+
+### `<ImportRemote />`
+
+```ts
+interface Props<Imported = any> {
+  source: string;
+  preload?: boolean;
+  onError(error: Error): void;
+  getImport(window: Window): Imported;
+  onImported(imported: Imported): void;
+}
+```
+
+#### Props
+
+**source**
+Source of the script to load the global from
+
+**preload**
+Generates a preload link tag for the source’s domain using [`@shopify/react-preconnect`](https://github.com/Shopify/quilt/tree/master/packages/react-preconnect)
```

Comment:
This is false. It generates a `preconnect` link, and not a `preload` link. https://developer.mozilla.org/en-US/docs/Web/HTML/Link_types

Reply:
@9at8 did you mean to fix this?

Reply:
I don't know if we're supposed to add preconnect links or preload links. I feel that we should do preload, but I'm not sure.

Reply:
both seem good tbh

lets make this reflect the code we have now though and we can do actual testing to see which is better diwn the line (which we should open issues for)